app/models/course/survey/answer_option.rb

```diff
@@ -3,4 +3,13 @@ class Course::Survey::AnswerOption < ActiveRecord::Base
   belongs_to :answer, inverse_of: :options
   belongs_to :question_option, class_name: Course::Survey::QuestionOption.name,
                                inverse_of: :answer_options
+
+  validate :question_option_belongs_to_question
+
+  private
+
+  def question_option_belongs_to_question
+    errors.add(:question_option_id, :invalid) \
+      unless answer.question.option_ids.include?(question_option_id)
```

Comment:
I think there's a N+1 where each answer loads the question and options upon validation, but I guess it's hard (and might not worth) to fix. One alternative is to filter out the invalid options in the params in controller.